### PR TITLE
Shadows example

### PR DESCRIPTION
**Pull request by kvark**
The multi-threading is implemented, but to get any performance benefit we need to generate way more objects. I was hoping for @csherratt help to replace my generation code by genmesh calls and make like a thousand of objects.

Shadows though seem to work good.
cc @fkaa @cmr

### CODE REVIEW

File: src/shadow/main.rs [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+                            let mvp = mx_view.mul_m(&ent.mx_to_world);
+                            mvp.into_fixed()
+                        };
+                        light.stream.draw(&batch).unwrap();
+                    }
+                    sender.send(light).unwrap();
+                })
+            }).collect();
+            // wait for them
+            drop(threads);
```
Comment (csherratt) [inline src/shadow/main.rs]: do you need to wait? you will block waiting on each light below. you might just want to avoid the extra delay.

Comment (kvark) [inline src/shadow/main.rs]: Right, fixed

**Comment by kvark** [comment]
Several things must be implemented to get a real benefit from multi-threading here:
  - increase the number of objects/cubes
  - increase the number of lights
  - remove allocations per draw/frame

Leaving it for the future PRs

**Comment by csherratt** [comment]
I am getting some pimples :(
strange

**Pull request merged**